packages/frontend/src/features/catalog/components/CatalogPanel.tsx

```diff
@@ -69,112 +84,217 @@ export const CatalogPanel: FC<React.PropsWithChildren<Props>> = ({
         [setSearch],
     );
 
-    /*
-    if (exploresResult.status === 'loading') {
-        return <LoadingSkeleton />;
-    }
+    const selectAndGetMetadata = useCallback(
+        (tableName: string) => {
+            // For optimization purposes, we could only make this request if metadata panel is open
+            setSelectedTable(tableName);
 
-    if (exploresResult.status === 'error') {
-        return (
-            <SuboptimalState
-                icon={IconAlertCircle}
-                title="Could not load explores"
-            />
-        );
-    }*/
+            if (catalogResults) {
+                const table = catalogResults.find(
+                    (item) => item.name === tableName,
+                );
+                if (table && table.type === CatalogType.Table)
+                    getMetadata(tableName);
+                else console.warn('Metadata not available for fields');
+            }
+        },
+        [catalogResults, getMetadata],
+    );
 
-    return (
-        <Stack>
-            <Box>
-                <Title order={4} mt="xxl">
-                    {projectData?.name}
-                </Title>
-                <Text color="gray">
-                    Select a table or field to start exploring.
-                </Text>
-            </Box>
-            <Group position="apart">
-                <TextInput
-                    w={'40%'}
-                    icon={<MantineIcon icon={IconSearch} />}
-                    rightSection={
-                        search ? (
-                            <ActionIcon onClick={() => setSearch('')}>
-                                <MantineIcon icon={IconX} />
-                            </ActionIcon>
-                        ) : null
+    const history = useHistory();
+    // Keyboard navigation
+    useHotkeys(
```

Comment:
Nice. I was worried this was table-based, but this should be portable to the tree